src/sentinel.c

```diff
@@ -1961,44 +2109,64 @@ void rewriteConfigSentinelOption(struct rewriteConfigState *state) {
         }
         dictReleaseIterator(di2);
     }
 
     /* sentinel current-epoch is a global state valid for all the masters. */
     line = sdscatprintf(sdsempty(),
         "sentinel current-epoch %llu", (unsigned long long) sentinel.current_epoch);
-    rewriteConfigRewriteLine(state,"sentinel",line,1);
+    rewriteConfigRewriteLine(state,"sentinel current-epoch",line,1);
 
     /* sentinel announce-ip. */
     if (sentinel.announce_ip) {
         line = sdsnew("sentinel announce-ip ");
         line = sdscatrepr(line, sentinel.announce_ip, sdslen(sentinel.announce_ip));
-        rewriteConfigRewriteLine(state,"sentinel",line,1);
+        rewriteConfigRewriteLine(state,"sentinel announce-ip",line,1);
     }
 
     /* sentinel announce-port. */
     if (sentinel.announce_port) {
         line = sdscatprintf(sdsempty(),"sentinel announce-port %d",
                             sentinel.announce_port);
-        rewriteConfigRewriteLine(state,"sentinel",line,1);
+        rewriteConfigRewriteLine(state,"sentinel announce-port",line,1);
     }
 
     /* sentinel sentinel-user. */
     if (sentinel.sentinel_auth_user) {
         line = sdscatprintf(sdsempty(), "sentinel sentinel-user %s", sentinel.sentinel_auth_user);
-        rewriteConfigRewriteLine(state,"sentinel",line,1);
+        rewriteConfigRewriteLine(state,"sentinel sentinel-user",line,1);
     }
 
     /* sentinel sentinel-pass. */
     if (sentinel.sentinel_auth_pass) {
         line = sdscatprintf(sdsempty(), "sentinel sentinel-pass %s", sentinel.sentinel_auth_pass);
-        rewriteConfigRewriteLine(state,"sentinel",line,1);
+        rewriteConfigRewriteLine(state,"sentinel sentinel-pass",line,1);
     }
 
 
     dictReleaseIterator(di);
+
+    /* mark rewrite as done for the optional configs */
+    rewriteConfigMarkAsProcessed(state,"sentinel monitor");
+    rewriteConfigMarkAsProcessed(state,"sentinel down-after-milliseconds");
+    rewriteConfigMarkAsProcessed(state,"sentinel failover-timeout");
+    rewriteConfigMarkAsProcessed(state,"sentinel parallel-syncs");
+    rewriteConfigMarkAsProcessed(state,"sentinel notification-script");
+    rewriteConfigMarkAsProcessed(state,"sentinel client-reconfig-script");
+    rewriteConfigMarkAsProcessed(state,"sentinel auth-pass");
+    rewriteConfigMarkAsProcessed(state,"sentinel auth-user");
+    rewriteConfigMarkAsProcessed(state,"sentinel config-epoch");
+    rewriteConfigMarkAsProcessed(state,"sentinel leader-epoch");
+    rewriteConfigMarkAsProcessed(state,"sentinel known-replica");
+    rewriteConfigMarkAsProcessed(state,"sentinel known-sentinel");
+    rewriteConfigMarkAsProcessed(state,"sentinel rename-command");
+    rewriteConfigMarkAsProcessed(state,"sentinel announce-ip");
+    rewriteConfigMarkAsProcessed(state,"sentinel announce-port");
+    rewriteConfigMarkAsProcessed(state,"sentinel sentinel-user");
+    rewriteConfigMarkAsProcessed(state,"sentinel sentinel-pass");
```

Comment:
i might be missing the point, but isn't it better to put all of these as an `else` for each of the above `if`s?
i.e. `rewriteConfigRewriteLine` already calls it, right? so we only need to do it when the `if` isn't entering.

even if that's not the case, can we at least put each of them next to the block that handles the rewrite of that config?
this way if we delete a block or clone it to create a similar one, we don't forget to update these too.

Reply:
Hi  @oranagra , I think it is proper to put these rewriteConfigMarkAsProcessed lines here IMHO. since for normal if wrapped configs (like sentinel announce-ip) we can put it into a else statements, there is no probllem. However for configs in loops we need a way to track whether it was processed, like sentinel parallel-syncs, if we do so this may not make te code looks clean. Maybe we can put comments which can clearly indicate we need to put rewriteConfigMarkAsProcessed  at then end of this function if we modify the configs. How do you think?

Reply:
then maybe for the majority of these you can put an `else` and for others put a comment indicating it is done elsewhere.
please note that there are some configs are missing from this block, like `myid`.
maybe that's on purpose, or an indication that this is risky this way, either way the intention should be clear by placing the line appropriately, or a comment instead.

Reply:
thanks @oranagra , I will put the others in else and keep loop wrapped one then. Some config missing is on purpose here, since for these one like `myid`, it will be processed by rewrite for sure, therefore we don't need to call rewriteConfigMarkAsProcessed at the end.

Reply:
Hi @oranagra , I added more comments in this part, and move others in an else statements, does this change looks good to you? thanks!